core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -530,15 +540,13 @@ public void validateIndexSettings(String indexName, Settings settings) throws In
             validationErrors.add("index must have 1 or more primary shards");
         }
         if (number_of_replicas != null && number_of_replicas < 0) {
-           validationErrors.add("index must have 0 or more replica shards");
-        }
-        if (validationErrors.isEmpty() == false) {
-            throw new IndexCreationException(new Index(indexName),
-                new IllegalArgumentException(getMessage(validationErrors)));
+            validationErrors.add("index must have 0 or more replica shards");
         }
+        return validationErrors;
+
     }
 
-    private String getMessage(List<String> validationErrors) {
+    public String getMessage(List<String> validationErrors) {
```

Comment:
Agree. I've added a `ValidationException` class and pushed a [commit](https://github.com/jasontedor/elasticsearch/commit/7135e14514a568177cd2a9a1038f083eeec83959) that reflects these suggestions.